big-red-shuttle/Controllers/DriverViewController.swift

```diff
@@ -1,17 +1,127 @@
+        
+        let cell = UITableViewCell(style: .default, reuseIdentifier: "cell")
+        if indexPath.row == 0 {
+            
+            let textField = UITextField(frame: CGRect(x: cell.bounds.width/2 + 20, y: 0, width: cell.bounds.width/2, height: cell.bounds.height))
+            cell.textLabel?.text = "Authentication Key"
+
+            if loginImage != nil {
+            cell.imageView?.image = loginImage
+            textField.text = textFieldText
+            }
+            textField.placeholder = "Insert Key"
+            textField.textAlignment = .right
+            textField.returnKeyType = .go
+            textField.isSecureTextEntry = true
+            textField.addTarget(self, action: #selector(textFieldDidChange), for: .editingDidEndOnExit)
+        cell.accessoryView = textField
+        } else {
+            if loginSuccessful {
+                cell.isUserInteractionEnabled = true
+                cell.textLabel?.textColor = .black
+            } else {
+                cell.isUserInteractionEnabled = false
+                cell.textLabel?.textColor = .lightGray
+                cell.textLabel?.alpha = 0.3
+            }
+            cell.textLabel?.text = "Log Location"
+            let logSwitch = UISwitch()
+            logSwitch.isOn = isLoggingOn
+            logSwitch.addTarget(self, action: #selector(didToggleLogging), for: .valueChanged)
+            cell.accessoryView = logSwitch
+        }
+        return cell
+    }
+    
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: false)
+    }
+    
+    
+    func tableView(_ tableView: UITableView, titleForFooterInSection section: Int) -> String? {
+        var footerString = ""
+        if loginSuccessful && isLoggingOn{
+            footerString = "You are now logging the location of your phone to the application and everyone on the network. Please turn off logging while you are not on the bus."
+        } else if loginSuccessful && !isLoggingOn {
+            footerString = "You are currently not logging the location of the bus. Click the toggle to start logging."
+        } else if !loginSuccessful {
+            footerString = "Your Authentication Key is incorrect. You are currently not logging the location."
+            }
+        
+        return footerString
+    }
+    
+    func textFieldDidChange(textField: UITextField) {
+        //check if access key is correct if so, unlock cell aka brsxappdev2016
+        if textField.text == "brsxappdev2016" {
+            loginImage = UIImage(named: "checkmark")
+            textFieldText = "brsxappdev2016"
+            loginSuccessful = true
+            tableView.reloadData()
+        } else {
+            loginImage = UIImage(named: "incorrect")
+            textFieldText = textField.text!
+            loginSuccessful = false
+            tableView.reloadData()
+        }
```

Comment:
call `tableView.reloadData()` after else statement since it's currently called in both the if and else statements